releasenotes/notes/discrete-rotations-5a77bda4ddd5e77e.yaml

```diff
@@ -0,0 +1,37 @@
+---
+features_transpiler:
+  - |
+    Added a new transpiler pass :class:`.DiscretizeRotations`,
+    that converts single-qubit :class:`.RZGate`, :class:`.RXGate` and :class:`.RYGate` rotations gates
+    whose angles are integer multiples of pi/4 into discrete sets of
+    Clifford, :class:`.TGate` and :class:`.TdgGate` gates.
+    Note that odd multiples of pi/4 require a single :class:`.TGate` and :class:`.TdgGate`,
+    as well as some Clifford gates,
+    while even multiples of pi/4, or equivalently, integer multiples of pi/2,
+    can be written using only Clifford gates.
+
+    For example::
+
+      from qiskit.circuit import QuantumCircuit
+      from qiskit.transpiler.passes import DiscretizeRotations
+      from qiskit.quantum_info import Operator
+
+      # The following quantum circuit consists of 5 Clifford gates
+      # and three single-qubit rotation gates whose angles are integer multiples of pi/4.
+      qc = QuantumCircuit(3)
+      qc.cx(0, 1)
+      qc.rz(pi/4, 0)
+      qc.cz(0, 1)
+      qc.rx(3*pi/4, 1)
+      qc.h(1)
+      qc.s(2)
+      qc.ry(2*pi/4, 2)
+      qc.cz(2, 0)
+
+      # The transformed circuit consists of Clifford, T and Tdg gates
+      qct = DiscretizeRotations()(qc)
+      clifford_t_names = get_clifford_gate_names() + ["t"] + ["tdg"]
+      assert(set(qct.count_ops().keys()).issubset(set(clifford_t_names)))
+
+      # The circuits before and after the transformation are equivalent
+      assert Operator(qc) == Operator(qct)
```

Comment:
I would suggest to move an example to the docstring of the class instead and point users there. The docs are the more natural location for an example, since the release notes are typically kept relatively short
```suggestion
```